nettrace/httpclient.go

```diff
@@ -58,16 +62,32 @@ type HTTPClient struct {
 	tracingStartedAt Timestamp
 	pendingTraces    *lockfree.Queue // value: networkTrace
 	noConnSockets    []*inetSocket   // not-yet connected AF_INET sockets
```

Comment:
Would it help to further decrease memory usage to turn `noConnSockets` also into `evictingMap` (with some suitable key)?